internal/pkg/admission/policy-server-deployment.go

```diff
@@ -118,13 +122,39 @@ func (r *Reconciler) updatePolicyServerDeployment(ctx context.Context, policySer
 	return nil
 }
 
-func shouldUpdatePolicyServerDeployment(originalDeployment *appsv1.Deployment, newDeployment *appsv1.Deployment) bool {
+func getPolicyServerImageFromDeployment(policyServer *v1alpha2.PolicyServer, deployment *appsv1.Deployment) (string, error) {
+	for containerIndex := range deployment.Spec.Template.Spec.Containers {
+		container := &deployment.Spec.Template.Spec.Containers[containerIndex]
+		if container.Name == policyServer.NameWithPrefix() {
+			return container.Image, nil
```

Comment:
doesn't this for just return the first policy-server image, as did the behaviour before this PR?

Reply:
It's iterating through all the containers, so it will return the right image. Before we were assuming the policy server was always the first container [Containers[0]](https://github.com/kubewarden/kubewarden-controller/blob/30d36df36827f27a4641b37b6caa663c20bd6404/internal/pkg/admission/policy-server-deployment.go#L123)

Reply:
doh, true, we just need "a" policy-server container image. Thanks!